penguin/web/services/projects.py

```
@@ -0,0 +1,355 @@
+    has_markdown_content = bool(markdown_content and markdown_content.strip())
+    if not has_spec_path and not has_markdown_content:
+        raise HTTPException(
+            status_code=400,
+            detail="Provide either spec_path or markdown_content to run a project workflow.",
+        )
+
+    if has_markdown_content:
+        markdown_source = str(markdown_content)
+        source_label = "inline_markdown"
+    else:
+        spec_file = Path(str(spec_path)).expanduser().resolve()
```

Ruff (0.15.11)

[warning] Async functions should not use pathlib.Path methods, use trio.Path or anyio.path

(ASYNC240)